lib/fortnox/api/repositories/base/json_convertion.rb

```diff
@@ -58,10 +67,32 @@ def sanitise( hash, keys_to_filter )
           end
         end
 
-        def remove_nil_values( hash )
-          hash.delete_if{ |_, value| value.nil? }
+        # Removes nil values and empty collections recursively from Hash
+        def clean_hash( hash )
+          cleaned_hash = hash.select{ |_, v| v != nil }
+          cleaned_hash.each do |k, v|
+            cleaned_hash[k] = clean_collection( v ) if v.respond_to?( :each )
+          end
         end
 
+        def clean_collection( collection )
+          case collection
+          when Array
+            collection = clean_array( collection )
+          when Hash
+            collection = clean_hash( collection )
+          end
+
+          collection
+        end
+
+        # Removes nil values and empty collections recursively from Array
+        def clean_array( array )
+          cleaned_array = array.select{ |e| e != nil }
+          cleaned_array.each_with_index do |e, i|
```

Comment:
```
cleaned_array.map{|e| e.respond_to?( :each ) ? clean_collection( e ) : e }`
```
Or even better, make `clean_collection` more general to `clean_value` and just add a check like this to the top of it:
```
return value unless value.respond_to?( :each )
```
That way you get this here instead:
```
cleaned_array.map{|e| clean_value( e )}`
```